plugins/gradle/testSources/org/jetbrains/plugins/gradle/properties/GradlePropertiesFileTest.kt

```diff
@@ -0,0 +1,108 @@
+// Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.plugins.gradle.properties
+
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.testFramework.LightPlatformTestCase
+import org.junit.Test
+import java.io.File
+import java.util.*
+import kotlin.io.path.Path
+
+class GradlePropertiesFileTest : LightPlatformTestCase() {
+
+  private val projectPath by lazy { project.basePath.orEmpty() }
+  private val externalProjectPath by lazy { Path(projectPath) }
+  private val gradlePropertiesPath by lazy { externalProjectPath.resolve(GRADLE_PROPERTIES_FILE_NAME).toString() }
+
+  override fun tearDown() {
+    super.tearDown()
+    closeAndDeleteProject()
```

Comment:
The reason behind was that the `LightPlatformTestCase` is reusing the same project directory between test execution, that's not a problem for the existing tests but in this particular one we are doing some modification to the project meaning that we want to have a clean directory between test executions. Right now `LightPlatformTestCase.tearDown()` isn't deleting the project directory for this reason was necessary to explicitly invoke the `closeAndDeleteProject()` method on `tearDown()`.

Here's a sample about this, when given a test with non present `local.properties` file, the test is failing because the file still exist from previous test execution, due that wasn't properly removed between test executions. 
<img width="1759" alt="directory with old test execution" src="https://user-images.githubusercontent.com/18151158/208946364-87744dc2-bf52-4756-a2c5-c3b9bebfca0d.png">

Reply:
Maybe it is enough to remove local.properties in the teardown phase? We will keep performance and stay explicit about the intention of the test. The file cleanup is more specific than project close and delete  (latter involves a lot of operations)

Reply:
That makes completely sense. The idea was to try to reuse some existing logic but as you mention this is overcomplicate a simple deletion